k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/GenericK8sLocatable.kt

```diff
@@ -0,0 +1,22 @@
+package com.amazon.spinnaker.keel.k8s.model
+
+import com.amazon.spinnaker.keel.k8s.*
+import com.netflix.spinnaker.keel.api.Locatable
+import com.netflix.spinnaker.keel.api.SimpleLocations
+
+interface GenericK8sLocatable : Locatable<SimpleLocations> {
+    val metadata: Map<String, String>
+    val template: K8sObjectManifest
+
+    val namespace: String
+        get() = (template.metadata[NAMESPACE] ?: NAMESPACE_DEFAULT) as String
+
+    override val application: String
+        get() = metadata.getValue(APPLICATION).toString()
+
+    override val id: String
+        get() = "$namespace-${template.kind}-${template.metadata[NAME]}".toLowerCase()
```

Comment:
Wonder if we can use spinnaker account name + kubernetes UID here somehow at least for vanilla k8s resources.

Reply:
k8s UID is only generated after we apply the resource to the cluster. I believe the keel resource `id` is required a priori, and as part of inserting it into the database. adding the account name is a good idea.